src/main/java/org/jenkinsci/plugins/artifactdeployer/ArtifactDeployerPublisher.java

```diff
@@ -68,6 +69,12 @@ public ArtifactDeployerPublisher(List<ArtifactDeployerEntry> deployedArtifact, b
     public Object readResolve() {
         if (this.entries == null)
             this.entries = Collections.emptyList();
+
+        if (Jenkins.getInstance().getPlugin("maven-project") != null)
```

Comment:
indeed. This is a temporary (sic) fix until I find some way to retrieve the parent job from this method to detect actual job type